controllers/cassandracluster/cassandra_status.go

```diff
@@ -52,17 +52,17 @@ func (rcc *CassandraClusterReconciler) updateCassandraStatus(ctx context.Context
 	}
 	needUpdate = false
 	//make also deepcopy to avoid pointer conflict
-	cc.Status = *status.DeepCopy()
-	err := rcc.Client.Status().Update(ctx, cc)
-	if err != nil {
-		logrus.WithFields(logrus.Fields{"cluster": cc.Name, "err": err}).Errorf("Issue when updating CassandraCluster Status")
-	}
```

Comment:
what's the reason for moving that part later down ?

Reply:
There is a problem that we need to update and set annotations first, otherwise line 55 gives error as annotation of updated object is nil.
The status is handling now a bit different.

Reply:
@cscetbon Does my explanation helps?